### DISCUSSION TITLE
"Connect" screen tweaks...

### DISCUSSION THREAD

**Discussion by QJKX** [jamulussoftware/jamulus#2194] [UI Related]
Change the window name from "Connect" to  "Server list" or "servers", because that better describes what it is.

Make a corresponding change on the main screen that renames "Connect" to "Server list" or "servers".
Now, if you're talking someone through it, there's only one button on screen that says "Connect" :-)

Remove the "Cancel" button - it's redundant.
Make Escape a keyboard shortcut to close the "servers" window (but not the main window).

The connect button has focus by default and will connect to a server if you push it - even if you didn't select a server yet.
Either it should be greyed out by default, or a server should be selected by default.
My preferred design would be for a server to be selected by default - then, by default you can use the arrow keys to change it, and enter to connect.
At the moment the initial focus is on the Directory drop down, and the arrow keys adjust the directory server.

Change "Server Name" to "Name" (The window is now already called "Servers" or "Server List").

Remove ms from each row, it's redundant and reduces information density.
Rename "Ping Time" to "Ping (ms)".

Use a monospace font for the number columns, and right align them - it makes stuff line up neatly reducing visual clutter.

Normalise the location information somehow, so it's better aligned and has some sorting value.
Perhaps there should be a Country column (that could show a flag) that's specified on the server using an ISO country code.
A blank default in the server config would be more useful than "yourCity". (I'm pretty much stealing the idea from xqf etc)

It's not obvious why some server names are bold and some are not.
Maybe server names should be bold when they aren't empty?
As a bonus a [uniwidth font](https://uxdesign.cc/uniwidth-typefaces-for-interface-design-b6e8078dc0f7) could be used, so the letters don't change position when they change weight. (Are there any suitable uniwidth fonts that are already installed on all platforms?)

From a usability point of view I don't want to have to choose a directory server.
Instead the genre should be a column in the server list. That's not really a UI problem.

Maybe there should be a "Song" or "subject" column too that users on the server can update. Unlikely they will, but it might give a better flavor of what the server is about. (it's akin to a 'map' on a game server browser). You might write "American Songbook", "Christmas Carols" "Blues in G" or "Jaja Ding Dong". Servers should probably blank it by default when all clients disconnect.

Should I attempt a PR?

**Comment by dcorson-ticino-com** [comment]
Before any PR this needs much discussion here.  Much of this has been discussed before.
The servers in bold are online more than 24 hours (if I remember correctly) to separate those that are for short term occasions.
Note that sorting geographically is pretty useless, what makes or breaks the deal is the ping time.  And the list is sorted by ping time.
I disagree that the connect button should be called a server list button.  When I open Jamulus I want to connect to other players, the server list may be the way to do that, but first my goal is to connect, not to have a list of servers.
Remember that the people using this are musicians, they are not interested in constantly adding info to the server list when they could just talk to the people connected and decide with them.  Maybe gamers would like that, but I don't think we will find many musicians.
Maybe using monospace fonts could improve the readability, but I haven't heard any complaints in the last year about that.
Let's see what others comment.

**Reply by QJKX** [reply]
>I disagree that the connect button should be called a server list button.

Well whatever it is, I think it should be consistent.
It's also called "Connection Setup" in the File menu :-)

**Reply by ann0see** [reply]
Maybe you can open an issue on changing connection setup to connect and state that you’d like to change it and open a PR. If nobody answers for a week, ping me.

I think in German it’s correctly phrased as "Connect", so I don’t see a Problem.

**Comment by npostavs** [comment]
> From a usability point of view I don't want to have to choose a directory server.
> Instead the genre should be a column in the server list. That's not really a UI problem.

I think this is a workaround to underlying technical problems. Something about relying on UDP messages to get the server list, and if the list is too long it won't fit in a single packet which messes things up, if I'm recalling correctly.

**Reply by chrisrimple** [reply]
There is a limit to the number of servers that can be assigned to a directory server, so there are multiple directory servers.  

But the idea of querying them all to build the server list, and showing the genre as a column in the server list, is interesting: instead of selecting a directory server (genre), the musician gets a list of all servers, so they see what's closest (by ping) including genre details.  That might help with the "can't find servers in use" problem, in that a musician won't have to check multiple directory servers to see what's active in that moment.

**Reply by rdica** [reply]
https://jamulus.live offers a single pane view of populated servers on the various Jamulus genres.

I do like the idea of something similar for the connect dialog BUT if I understand correctly, this would also increase network usage substantially as every server is pinged to determine ping time. Not ideal.

**Comment by rdica** [comment]
> The connect button has focus by default and will connect to a server if you push it - even if you didn't select a server yet.
> Either it should be greyed out by default, or a server should be selected by default.

And for users that direct connect, the address or IP is saved in the section where its typed in, why would we want some default server highlighted, overriding the saved server, when it may never be connected?

In my testing, I removed the saved server I direct connect to, ensured there was no highlighted server, then hit Connect, and indeed a server was connected to.
I feel thats a bug.
The server that got connected to wasn't even the one closest to me based on ping time, but was in Frankfurt Germany, called "Directory Server" in Any Genre1.
Is that a program default?

I have no issues with layout and appearance. Improved readability is always a plus.
My use case for the connection dialog is roughly 50-50 direct connect and server select.

**Reply by ann0see** [reply]
> Is that a program default?

Yes it is - and highly confusing. There was a discussion about it a few weeks ago, but we didn’t decide on how to handle it yet…

@QJKX BTW: thanks for all your suggestions. I think you have a Good summary of places where the UI is confusing. Good to have them summed up!

**Reply by QJKX** [reply]
@ann0see  Glad it's of some use.

**Comment by ann0see** [comment]
> The connect button has focus by default and will connect to a server if you push it - even if you didn't select a server yet.
Either it should be greyed out by default, or a server should be selected by default.

 I assume many people feel this is a bug, so I‘d also suggest to open a feature request in the issues tab and wait for a discussion. You can ping softins and me there, I think?

**Comment by ann0see** [comment]
> Use a monospace font for the number columns, and right align them - it makes stuff line up neatly reducing visual clutter.

… and all the other suggestion about UI: could you please post a draft? I assume @jamulussoftware/designers should look into it.
